crates/biome_css_parser/src/syntax/mod.rs

```diff
@@ -601,3 +601,83 @@ pub(crate) fn parse_string(p: &mut CssParser) -> ParsedSyntax {
 fn is_at_string(p: &mut CssParser) -> bool {
     p.at(CSS_STRING_LITERAL)
 }
+
+/// Attempt to parse some input with the given parsing function. If parsing
+/// succeeds, `Ok` is returned with the result of the parse and the state is
+/// preserved. If parsing fails, this function rewinds the parser back to
+/// where it was before attempting the parse and the `Err` value is returned.
+#[allow(dead_code)] // TODO: Remove this allow once it's actually used
+pub(crate) fn try_parse<T, E>(
+    p: &mut CssParser,
+    func: impl FnOnce(&mut CssParser) -> Result<T, E>,
+) -> Result<T, E> {
+    let checkpoint = p.checkpoint();
```

Comment:
Do we need to implement `speculative_parsing`?
It blocks a parse recovery since we're going to rewind anyway.

Reply:
I suppose it's probably best that we do just for safety. I don't think we _need_ it strictly for how it'll be used here, but definitely no harm in including it. I'll add it in now.